src/pd/Client.cc

```diff
@@ -214,16 +218,26 @@ void Client::switchLeader(const ::google::protobuf::RepeatedPtrField<std::string
 
 void Client::updateURLs(const ::google::protobuf::RepeatedPtrField<::pdpb::Member> & members)
 {
-    std::vector<std::string> tmp_urls;
+    std::deque<std::string> tmp_urls;
     for (const auto & member : members)
     {
         auto client_urls = member.client_urls();
         for (auto & client_url : client_urls)
         {
-            tmp_urls.push_back(client_url);
+            // Check if the URL is in the failed_urls list
+            if (failed_urls.count(client_url) > 0)
+            {
+                // If it is, add it to the end of the list
+                tmp_urls.push_back(client_url);
+            }
+            else
+            {
+                // If it is not, add it to the front of the list
+                tmp_urls.push_front(client_url);
```

Comment:
This will change the order of normal URLs. So when each time `updateLeader` is called, the order of URLs will not be the same. Does it have some side effects?

Reply:
I think no, `urls` is used to get all members of pd, and request to any pd can get the correct result.